routes/api/collectors.js

```diff
@@ -0,0 +1,64 @@
+var express = require('express');
+var router = express.Router();
+var gcloud = require('gcloud');
+
+var datastore = gcloud.datastore({
+  projectId: 'prototype-content-consumer',
+  keyFilename: './credentials/private-key.json'
+});
+
+/* fields => {
+               productname: chair,
+               hostname: www.ikea.com,
+               pageurl: http://www.ikea.com/us/en/catalog/products/S89100143/,
+               ...
+             }
+*/
+
+function saveCollector(fields) {
+  let collectorKey = datastore.key('Collector');
+  // TODO: created at / updated at fields
+  datastore.save({
+    key: collectorKey,
+    data: fields
+  }, function (err) {
+    if (err) {
+      return res.json({errors: req});
+    }
+  });
+}
+
+/* POST /api/v1/collectors
+   body: {
+    hostname=www.ikea.com,
+    productname=sweater,
+    pageurl=http://www.ikea.com/us/en/catalog/products/S89100143/
+ }
+*/
+
+router.post('/', function(req, res, next) {
+  saveCollector(req.body)
+  if (req.body) {
+    res.json({
+      "data": req.body,
+      "jsonapi": {
+        "version": "1.0.0"
+      }
+    });
+  } else {
```

Comment:
The validation should be more specific than only the body. This can be off-loaded into the model as well as a validation mechanism before persistence. As a starter we can create savePayload inside a models.js and pass the request.body and expect all products to have a structure that is similar. A validateProduct method can exist within the models.js too, and savePayload will call that first before persisting to gcloud datastore. If validation fails, the savePayload can call a errorCallback with the set of errors. Let's not worry about the exact field validation to start, just put the plumbing into the code and we'll get more strict as we learn more. As a starter, maybe validateProduct just returns true.

import models from './models'
// let models = require('./models');

models.savePayload(request.body, function(object) {
  res.json(object);
}, function(errors) {
  res.json(errors);
});